clouddriver-kubernetes/src/test/resources/com/netflix/spinnaker/clouddriver/kubernetes/v2/op/handler/statefulsetpartitionbase.json

```diff
@@ -0,0 +1,151 @@
+                            "nodeSelectorTerms": [
+                                {
+                                    "matchExpressions": [
+                                        {
+                                            "key": "node-role.kubernetes.io/testapp",
+                                            "operator": "Exists"
+                                        }
+                                    ]
+                                }
+                            ]
+                        }
+                    }
+                },
+                "containers": [
+                    {
+                        "image": "testapp/$image:",
+                        "imagePullPolicy": "IfNotPresent",
+                        "livenessProbe": {
+                            "failureThreshold": 3,
+                            "httpGet": {
+                                "path": "/",
+                                "port": 5000,
+                                "scheme": "HTTP"
+                            },
+                            "initialDelaySeconds": 20,
+                            "periodSeconds": 10,
+                            "successThreshold": 1,
+                            "timeoutSeconds": 2
+                        },
+                        "name": "testapp",
+                        "ports": [
+                            {
+                                "containerPort": 5000,
+                                "name": "http",
+                                "protocol": "TCP"
+                            }
+                        ],
+                        "readinessProbe": {
+                            "failureThreshold": 3,
+                            "httpGet": {
+                                "path": "/",
+                                "port": 5000,
+                                "scheme": "HTTP"
+                            },
+                            "initialDelaySeconds": 20,
+                            "periodSeconds": 10,
+                            "successThreshold": 1,
+                            "timeoutSeconds": 2
+                        },
+                        "resources": {},
+                        "terminationMessagePath": "/dev/termination-log",
+                        "terminationMessagePolicy": "File",
+                        "volumeMounts": [
+                            {
+                                "mountPath": "/test",
+                                "name": "$name"
+                            }
+                        ]
+                    }
+                ],
+                "dnsPolicy": "ClusterFirst",
+                "restartPolicy": "Always",
+                "schedulerName": "default-scheduler",
+                "securityContext": {},
+                "terminationGracePeriodSeconds": 10,
+                "tolerations": []
+            }
+        },
+        "updateStrategy": {
```

Comment:
Do we also have a test for the case where `updateStrategy` isn't set? I'm pretty sure it gets a default value but it would be good to know that `statefulSet.getSpec().getUpdateStrategy().getRollingUpdate();` never throws an NPE.

Reply:
Great question. I will add a case for it because as you state, I was also pretty sure it gets a default value, but this would code would definitely NPE if `updateStrategy` is missing.

Reply:
FWIW I did some manual testing around this, and the statefulset always has `updateStrategy` regardless of if it's created/patched with that section or not. I think this is safe. Also pretty sure we would have seen problems in the past, since https://github.com/spinnaker/clouddriver/blob/d455405a66389b2b27f0d0b0ce82c020fa348810/clouddriver-kubernetes/src/main/groovy/com/netflix/spinnaker/clouddriver/kubernetes/v2/op/handler/KubernetesStatefulSetHandler.java#L117-L119 should have NPE'd before now. :)